### PR TITLE
poll for wallet web3 availability

### PR DESCRIPTION
**Pull request by cloudonshore**
Sometimes a web3 provider can inject their `window.web3` object later than expected, or a user can turn it on after the page loads. This PR helps with "liveness" of that state.

### CODE REVIEW

File: src/wallet/redux/middleware.js [hunk 1]
```diff
@@ -178,69 +178,61 @@ function connectFortmatic(store) {
...
-        type: 'SET_WALLET_AVAILABILITY',
-        wallets: walletsAvailable,
-      })
-    }
   }
 }
 
 export default function walletMiddleware(store) {
   detectWeb3Wallets(store)
+  window.setInterval(() => detectWeb3Wallets(store), 5000)
```
Comment (syjk129) [inline src/wallet/redux/middleware.js]: Do you need to clear the interval once it finds the web3 wallet?

Comment (cloudonshore) [reply src/wallet/redux/middleware.js]: No the idea is to keep liveness, I also want to disable their ability to connect if they disable the wallet after they find it. The state could also be used in the future to prompt a window like `web3 connection lost, please re-enable your wallet` if the user disabled their extension in the middle of a session, or if the extension crashed for some reason.

Comment (perich) [reply src/wallet/redux/middleware.js]: I thought maybe this was intentional in case the user disabled/re-enables the extension or something

Comment (cloudonshore) [reply src/wallet/redux/middleware.js]: yep correct @perich

Comment (syjk129) [reply src/wallet/redux/middleware.js]: Ah that makes sense 👍

**Pull request merged**